src/Microsoft.TemplateEngine.Edge/Settings/GlobalSettings.cs

```diff
@@ -1,42 +1,198 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+#nullable enable
+
 using System;
 using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.TemplateEngine.Abstractions;
 using Microsoft.TemplateEngine.Abstractions.GlobalSettings;
 using Newtonsoft.Json;
 
 namespace Microsoft.TemplateEngine.Edge.Settings
 {
-    public class GlobalSettings : IGlobalSettings
+    class GlobalSettings : IGlobalSettings, IDisposable
     {
-        [JsonProperty("UserInstalledTemplatesSources")]
-        private List<TemplatesSourceData> _userInstalledTemplatesSources = new List<TemplatesSourceData>();
+        private readonly Paths _paths;
+        private readonly IEngineEnvironmentSettings _environmentSettings;
+        private readonly string _globalSettingsFile;
+        private bool _locked;
+        private IDisposable _watcher;
+
+        public GlobalSettings(IEngineEnvironmentSettings environmentSettings, string globalSettingsFile)
+        {
```

Comment:
public members should check if Dispose has been called & if so, throw https://docs.microsoft.com/en-us/dotnet/api/system.objectdisposedexception?view=net-5.0

Reply:
applicable to all public members

Reply:
I added
```
            if (_disposed)
            _userInstalledTemplatesSources.Add(userInstalledTemplate);
            {
            SettingsChanged?.Invoke();
                throw new ObjectDisposedException(nameof(GlobalSettings));
            }
```
To all public methods